pkg/apis/kops/validation/aws.go

```diff
@@ -107,3 +109,15 @@ func awsValidateAMIforNVMe(fieldPath *field.Path, ig *kops.InstanceGroup) field.
 	}
 	return allErrs
 }
+
+func awsValidateSpotDurationInMinute(fieldPath *field.Path, ig *kops.InstanceGroup) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if ig.Spec.SpotDurationInMinutes != nil {
+		validSpotDurations := map[int64]struct{}{60: {}, 120: {}, 180: {}, 240: {}, 300: {}, 360: {}}
+		if _, ok := validSpotDurations[*ig.Spec.SpotDurationInMinutes]; !ok {
+			errDetails := fmt.Sprintf("must be less than 360, greater than 60 and be a multiple of 60")
+			allErrs = append(allErrs, field.Invalid(fieldPath, ig.Spec.SpotDurationInMinutes, errDetails))
+		}
+	}
```

Comment:
Could you try using IsValidValue like here? It should work if you convert the value to string first.
https://github.com/kubernetes/kops/blob/03adc95c4545ccd1c7c1d565d7c3a1a0b2747294/pkg/apis/kops/validation/validation.go#L440-L443

Reply:
Done ✅